sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.targets

```diff
@@ -33,9 +33,10 @@ WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and
         <_FunctionsExtensionsFullPublish Condition="$(_FunctionsExtensionsFullPublish) == ''">!$(NoBuild)</_FunctionsExtensionsFullPublish>
         <MSBuildFunctionsTargetsPath>$(MSBuildExtensionsPath)\Microsoft\VisualStudio\Managed.Functions\</MSBuildFunctionsTargetsPath>
       
-        <FunctionsEnableWorkerIndexing Condition="$(FunctionsEnableWorkerIndexing) == ''">false</FunctionsEnableWorkerIndexing>
-        <FunctionsMetadataSourceGen_Enabled Condition="$(FunctionsMetadataSourceGen_Enabled) == ''">true</FunctionsMetadataSourceGen_Enabled>
         <FunctionsEnablePlaceholder Condition="$(FunctionsEnablePlaceholder) == ''">false</FunctionsEnablePlaceholder>
+        <FunctionsEnableWorkerIndexing Condition="$(FunctionsEnablePlaceholder) == 'true'">true</FunctionsEnableWorkerIndexing>
+        <FunctionsEnableWorkerIndexing Condition="$(FunctionsEnableWorkerIndexing) == '' And !$(FunctionsEnablePlaceholder)">false</FunctionsEnableWorkerIndexing>
+        <FunctionsEnableMetadataSourceGen Condition="$(FunctionsEnableWorkerIndexing) == 'true' And $(FunctionsEnableMetadataSourceGen) == ''">true</FunctionsEnableMetadataSourceGen>
```

Comment:
Another question - do we want to forcibly set `FunctionsEnableMetadataSourceGen` to false if `FunctionsEnableWorkerIndexing` is not enabled?

Reply:
> Another question - do we want to forcibly set `FunctionsEnableMetadataSourceGen` to false if `FunctionsEnableWorkerIndexing` is not enabled?

Do we need both the settings? Why not just "FunctionsEnableWorkerIndexing"?

Reply:
https://github.com/Azure/azure-functions-dotnet-worker/issues/1155 - `FunctionsEnableMetadataSourceGen` was introduced so people could turn off generation in cases of compilation errors. 

There is a path for worker-indexing without source-generation (reads `functions.json` after worker initialization) which would be taken if worker indexing was true and source generation was false. Not sure which scenarios someone would have to use this.

If we want worker indexing to _only_ use source-generation, we can cut down to just one prop.

Reply:
got it. May be we can re-evaluate this in few months again?

Reply:
Will add #2 - I think 3 can come in a follow up.